src/binding/python/RecordComponent.cpp

```diff
@@ -425,7 +440,25 @@ load_chunk(RecordComponent & r, py::tuple const & slices)
     std::vector<bool> flatten;
     std::tie(offset, extent, flatten) = parseTupleSlices(ndim, full_extent, slices);
 
-    return load_chunk(r, offset, extent, flatten);
+    // some one-size dimensions might be flattended in our output due to selections by index
+    size_t const numFlattenDims = std::count(flatten.begin(), flatten.end(), true);
+    std::vector< ptrdiff_t > shape(extent.size() - numFlattenDims);
+    auto maskIt = flatten.begin();
+    std::copy_if(
+            std::begin(extent),
+            std::end(extent),
+            std::begin(shape),
+            [&maskIt](std::uint64_t){
+                return !*(maskIt++);
+            }
+    );
+
+    auto const dtype = dtype_to_numpy( r.getDatatype() );
+    auto a = py::array( dtype, shape );
```

Comment:
Note from [here](https://github.com/openPMD/openPMD-api/pull/912#issuecomment-769739567): even in the original code, users can delete the returned object (or not catch it at all) which results in a use-after-free in flush.

Check that this becomes safe with this PR.